File: src/VisualBasic/CodeCracker/Usage/MethodAnalyzers/MethodInformation.vb
Namespace Usage.MethodAnalyzers
    Public Class MethodInformation
        Public Property MethodName As String
        Public Property MethodFullDefinition As String
        Public Property ArgumentIndex As Integer
        Public Property MethodAction As Action(Of List(Of Object))

        Public Sub New(methodName As String, methodFullDefinition As String, methodAction As Action(Of List(Of Object)), Optional argumentIndex As Integer = 0)
            Me.MethodName = methodName
            Me.MethodFullDefinition = methodFullDefinition
            Me.ArgumentIndex = argumentIndex
            Me.MethodAction = methodAction
        End Sub
    End Class
End Namespace

File: src/CSharp/CodeCracker/Usage/MethodAnalyzers/MethodInformation.cs
using System;
using System.Collections.Generic;

namespace CodeCracker.CSharp.Usage.MethodAnalyzers
{
    public class MethodInformation
    {
        public string MethodName { get; set; }
        public string MethodFullDefinition { get; set; }
        public int ArgumentIndex { get; set; }
        public Action<List<object>> MethodAction { get; set; }

        public MethodInformation(string methodName, string methodFullDefinition, Action<List<object>> methodAction, int argumentIndex = 0)
        {
            MethodName = methodName;
            MethodFullDefinition = methodFullDefinition;
            ArgumentIndex = argumentIndex;
            MethodAction = methodAction;
        }
    }
}